apps/hash-frontend/src/pages/[shortname].page/profile-bio.tsx

```diff
@@ -0,0 +1,371 @@
+import { IconButton, PenRegularIcon } from "@hashintel/design-system";
+import { TextToken } from "@local/hash-graphql-shared/graphql/types";
+import { paragraphBlockComponentId } from "@local/hash-isomorphic-utils/blocks";
+import { types } from "@local/hash-isomorphic-utils/ontology-types";
+import { ProfileBioProperties } from "@local/hash-isomorphic-utils/system-types/shared";
+import {
+  EntityId,
+  EntityRootType,
+  OwnedById,
+  Subgraph,
+} from "@local/hash-subgraph/.";
+import { getOutgoingLinkAndTargetEntities } from "@local/hash-subgraph/stdlib";
+import { extractBaseUrl } from "@local/hash-subgraph/type-system-patch";
+import { Box, Skeleton, Typography } from "@mui/material";
+import {
+  FunctionComponent,
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from "react";
+
+import { BlockCollection } from "../../blocks/block-collection/block-collection";
+import { BlockLoadedProvider } from "../../blocks/on-block-loaded";
+import { UserBlocksProvider } from "../../blocks/user-blocks";
+import { useBlockProtocolCreateEntity } from "../../components/hooks/block-protocol-functions/knowledge/use-block-protocol-create-entity";
+import { useBlockProtocolGetEntity } from "../../components/hooks/block-protocol-functions/knowledge/use-block-protocol-get-entity";
+import { BlockCollectionContentItem } from "../../graphql/api-types.gen";
+import { Org, User } from "../../lib/user-and-org";
+import { CheckRegularIcon } from "../../shared/icons/check-regular-icon";
+import { GlobeRegularIcon } from "../../shared/icons/globe-regular-icon";
+import { ProfileSectionHeading } from "../[shortname]/shared/profile-section-heading";
+
+const getProfileBioContents = (params: {
+  profileBioSubgraph: Subgraph<EntityRootType>;
+  profileBioEntityId: EntityId;
+}): BlockCollectionContentItem[] => {
+  const { profileBioEntityId, profileBioSubgraph } = params;
+  const outgoingContentLinks = getOutgoingLinkAndTargetEntities(
+    profileBioSubgraph,
+    profileBioEntityId,
+  )
+    .filter(
+      ({ linkEntity: linkEntityRevisions }) =>
+        linkEntityRevisions[0] &&
+        linkEntityRevisions[0].metadata.entityTypeId ===
+          types.linkEntityType.contains.linkEntityTypeId,
+    )
+    .sort((a, b) => {
+      const aLinkEntity = a.linkEntity[0]!;
+      const bLinkEntity = b.linkEntity[0]!;
+
+      return (
+        (aLinkEntity.linkData?.leftToRightOrder ?? 0) -
+          (bLinkEntity.linkData?.leftToRightOrder ?? 0) ||
+        aLinkEntity.metadata.recordId.entityId.localeCompare(
+          bLinkEntity.metadata.recordId.entityId,
+        ) ||
+        aLinkEntity.metadata.temporalVersioning.decisionTime.start.limit.localeCompare(
+          bLinkEntity.metadata.temporalVersioning.decisionTime.start.limit,
+        )
+      );
+    });
```

Comment:
This looks duplicative of `getBlockCollectionBlocks` in the API – I guess it's because we use a special-cased `page` query for pages, but not here?

Reply:
Yes, I didn't think it makes sense to introduce a `BlockCollection` query with the `contents` resolvers in the same way we currently have it for pages, as it's my understanding we want to move away from these custom field resolvers and do more of this work in the FE. This function should probably be moved somewhere else when we no longer make use of field resolvers to query pages, but not sure when we'll get around to refactoring this.